engine/language-server/src/main/scala/org/enso/languageserver/requesthandler/executioncontext/GetComponentGroupsHandler.scala

```diff
@@ -0,0 +1,89 @@
+package org.enso.languageserver.requesthandler.executioncontext
+
+import akka.actor.{Actor, ActorRef, Cancellable, Props}
+import com.typesafe.scalalogging.LazyLogging
+import org.enso.jsonrpc._
+import org.enso.languageserver.data.ClientId
+import org.enso.languageserver.requesthandler.RequestTimeout
+import org.enso.languageserver.runtime.ExecutionApi._
+import org.enso.languageserver.runtime.{
+  ContextRegistryProtocol,
+  RuntimeFailureMapper
+}
+import org.enso.languageserver.util.UnhandledLogging
+
+import scala.concurrent.duration.FiniteDuration
+
+/** A request handler for `executionContext/getComponentGroups` commands.
+  *
+  * @param timeout request timeout
+  * @param contextRegistry a reference to the context registry.
+  * @param clientId a client connected to the language server
+  */
+class GetComponentGroupsHandler(
```

Comment:
It is a bit confusing that we have two objects called `GetComponentGroupsHandler`.
For a moment when I was reading a code I saw that `GetComponentGroupsHandler` forwards the request to the `ConextRegistry` and then `ContextRegistry` forwards the request to `GetComponentGroupsHandler` -  I was confused - "an infinite loop?!". After careful reading it turned out it is a different class though.

I may be biased - if I opened this in the IDE and could navigate to sources of each I may not even notice it, but when looking at this code on GitHub where I cannot easily Go-To-Definition, I got quite confused. Maybe we could solve this by giving them slightly different names? They do slightly differnt things after all.